Purchases/Public/RCPurchases.m

```diff
@@ -223,13 +266,71 @@ - (void)setDelegate:(id<RCPurchasesDelegate>)delegate
 #pragma mark - Public Methods
 
 #pragma mark Attribution
+
+- (void)addAttributionData:(NSDictionary *)data
+               fromNetwork:(RCAttributionNetwork)network
+{
+    [self addAttributionData:data fromNetwork:network forNetworkUserId:nil];
+}
+
 - (void)addAttributionData:(NSDictionary *)data
                fromNetwork:(RCAttributionNetwork)network
+          forNetworkUserId:(NSString * _Nullable)networkUserId
 {
-    if (data.count > 0) {
-        [self.backend postAttributionData:data
-                              fromNetwork:network
-                             forAppUserID:self.appUserID];
+    NSString *networkKey = [NSString stringWithFormat:@"%ld",(long)network];
+    NSString *advertisingIdentifier = [self.attributionFetcher advertisingIdentifier];
+    NSString *newValueForNetwork = [NSString stringWithFormat:@"%@_%@", advertisingIdentifier, networkUserId];
+    NSString *cacheKey = [self attributionDataUserDefaultCacheKeyForAppUserID:self.appUserID];
+    NSDictionary *dictOfLatestSentToNetworks = [self.userDefaults objectForKey:cacheKey];
+    
+    NSString *latestSentToNetwork = dictOfLatestSentToNetworks[networkKey];
+    
+    if (latestSentToNetwork && [latestSentToNetwork isEqualToString:newValueForNetwork]) {
+        RCDebugLog(@"Attribution data is the same as latest. Skipping.");
+    } else {
+        NSMutableDictionary<NSString *, NSString *> *newDictToCache = [NSMutableDictionary dictionaryWithDictionary:dictOfLatestSentToNetworks];
+        newDictToCache[networkKey] = newValueForNetwork;
+
+        NSMutableDictionary *newData = [NSMutableDictionary dictionaryWithDictionary:data];
+        newData[@"rc_idfa"] = advertisingIdentifier;
+        newData[@"rc_idfv"] = [self.attributionFetcher identifierForVendor];
+        newData[@"rc_attribution_network_id"] = (networkUserId != nil) ? networkUserId : self.appUserID;
```

Comment:
another cool hack!
`newData[@"rc_attribution_network_id"] = networkUserId ?: self.appUserID;`

does the same thing